share/spice/npm/npm.js

```diff
@@ -44,22 +45,23 @@
                         value: item.repository.url
                     });
                 }
-                
+
                 if (item.engines) {
                     boxData.push({
                         label: "Engines",
                         value: item.engines.node
                     });
                 }
-                
+
                 if (item.dist) {
                     boxData.push({
                         label: "Source",
-                        value: item.dist.tarball
+                        value: item.dist.tarball,
+                        url: item.dist.tarball
                     });
                 }
 
-                if (item.dependencies) {      
+                if (Object.keys(item.dependencies).length > 0) {
```

Comment:
The problem is, in case of no dependency, `dependencies` does exist but has no key or value. The only way to check if it's empty is to check the number of keys.